packages/firestore/src/core/sync_engine.ts

```diff
@@ -240,19 +240,19 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
    * snapshot.
    */
   private async initializeViewAndComputeSnapshot(
-    queryData: QueryData,
+    targetData: TargetData,
     current: boolean
   ): Promise<ViewSnapshot> {
-    const query = queryData.query;
-    const docs = await this.localStore.executeQuery(query);
+    const target = targetData.target;
+    const docs = await this.localStore.executeTargetQuery(target);
```

Comment:
Per chat conversation, I think we should never "execute" a `Target` because `Target` represents a server query and so we can send it to the backend and get the server results, or we can look up cached server results from the targetDocuments table in persistence, but we should not execute it directly.  Instead our local query execution should happen on `Query` objects.

That may mean we need to refactor this method so it takes a `Query` and a `targetId` instead of `TargetData`.